### PR TITLE
Fix of ITS hit generation (instead of #312)

### PR DESCRIPTION
**Pull request by shahor02**
1) Adopt new BaseHit::BaseXYZE_f_f as a base class for ITS hit
2) Make sure just 1 hit is generated at each traversal of sensor by track
3) Lot of cleaning in ITS/Detector
4) Squashed separate commit to make ITSMFT Point c-tor inline

The previous PR #312 became polluted due, reopening here after squashing 2 relevant commits

### CODE REVIEW

File: Detectors/ITSMFT/ITS/simulation/include/ITSSimulation/Detector.h [hunk 1]
```diff
@@ -255,39 +244,18 @@ class Detector : public AliceO2::Base::Detector
...
-    Double32_t mTime;                 //! time
-    Double32_t mLength;               //! length
-    Double32_t mEnergyLoss;           //! energy loss
-
+    /// this is transient data about track passing the sensor
+    struct TrackData {                  // this is transient 
+      bool  mHitStarted;                //! hit creation started
+      unsigned char mTrkStatusStart;    //! track status flag
+      TLorentzVector mPositionStart;    //! position at entrance
+      TLorentzVector mMomentumStart;    //! momentum
```
Comment (sawenzel) [inline Detectors/ITSMFT/ITS/simulation/include/ITSSimulation/Detector.h]: Are these really 4-vectors or rather Point3D + Vector3D ? ... We can potentially fix this later (if we create an issue)

**Comment by shahor02** [comment]
@sawenzel: TLorentzVector is TVector3 + 1 double

On 6 Apr 2017 7:59 am, sawenzel <notifications@github.com> wrote:

@sawenzel commented on this pull request.

________________________________

In Detectors/ITSMFT/ITS/simulation/include/ITSSimulation/Detector.h<https://github.com/AliceO2Group/AliceO2/pull/313#discussion_r110083607>:

> -    Int_t mShunt;                     //! shunt
-    Int_t mTrkStatusFlag;             //! track status flag
-    TLorentzVector mPosition;         //! position
-    TLorentzVector mEntrancePosition; //! position at entrance
-    TLorentzVector mMomentum;         //! momentum
-    Double32_t mEntranceTime;         //! time at entrance
-    Double32_t mTime;                 //! time
-    Double32_t mLength;               //! length
-    Double32_t mEnergyLoss;           //! energy loss
-
+    /// this is transient data about track passing the sensor
+    struct TrackData {                  // this is transient
+      bool  mHitStarted;                //! hit creation started
+      unsigned char mTrkStatusStart;    //! track status flag
+      TLorentzVector mPositionStart;    //! position at entrance
+      TLorentzVector mMomentumStart;    //! momentum


Are these really 4-vectors or rather Point3D + Vector3D ? ... We can potentially fix this later (if we create an issue)

**Comment by sawenzel** [comment]
@shahor02 : I think you misunderstood. I was just wondering whether eventually `mPositionStart` is a `Position3D<float>` and mMomentumStart is a `Vector3D<float>` or similar. 

Merging for now.

**Comment by shahor02** [comment]
@sawenzel : these are really 2 4-vectors, as filled by TVirtualMC::GetMC()->TrackMomentum, TrackPosition, but the time component of mEntrancePosition is not used, instead I filled the time of 
exit position vector, which is the local TLorentzVector positionStop in Detector::ProcessHits

**Pull request merged**